src/main/java/com/manywho/services/saml/services/JwtService.java

```diff
@@ -1,26 +1,61 @@
 package com.manywho.services.saml.services;
 
-import com.auth0.jwt.JWTSigner;
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.JWTVerifier;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.exceptions.JWTVerificationException;
+import com.auth0.jwt.interfaces.DecodedJWT;
 import org.joda.time.DateTime;
-
 import javax.inject.Inject;
-import java.util.HashMap;
-import java.util.Map;
 
 public class JwtService {
-    private final JWTSigner jwtSigner;
+    private Algorithm algorithm;
+    private JWTVerifier verifier;
 
     @Inject
-    public JwtService(JWTSigner jwtSigner) {
-        this.jwtSigner = jwtSigner;
+    public JwtService(String secret) {
+        algorithm = Algorithm.HMAC256(secret);
+        verifier = JWT.require(algorithm)
+                .withIssuer("saml-service")
+                .acceptLeeway(1)   //1 sec for nbf and iat
+                .acceptExpiresAt(5)   //5 secs for exp
```

Comment:
I don't know enough about the SAML, but isn't too restrictive? As a small network lag could easily take more than 5 seconds.
Also the nbf of only 1 second could be too restrictive IMHO

Reply:
I don't think it should be a problem, those values are usually representing differences in the internal clock of the Service Provider and Identity Provider. But there are already http redirects that would make those values not important at all. The expiration time in the assertion should also be a couple of minutes so it shouldn't be either a problem.